### PR TITLE
make compatible with laravel 5.6

### PR DESCRIPTION
**Pull request by mrabbani**

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -3,9 +3,8 @@
     "description": "Elastic Driver for Laravel Scout",
     "keywords": ["laravel", "scout", "elasticsearch", "elastic"],
     "require": {
-        "php": ">=5.6.4",
-        "laravel/scout": "^3.0|^4.0",
-        "elasticsearch/elasticsearch": "^5.0"
+        "laravel/scout": "^5.0",
+        "elasticsearch/elasticsearch": "^6.0"
```
Comment (kronthto) [inline composer.json:7]: Is upgrading to ES-PHP 6 really required for Laravel 5.6 / Scout 5? Maybe this should be a seperate PR.

File: src/ElasticsearchEngine.php [hunk 2]
```diff
@@ -200,32 +179,29 @@ public function mapIds($results)
...
      * Map the given results to instances of the given model.
      *
+     * @param  \Laravel\Scout\Builder  $builder
      * @param  mixed  $results
      * @param  \Illuminate\Database\Eloquent\Model  $model
-     * @return Collection
+     * @return \Illuminate\Database\Eloquent\Collection
      */
-    public function map($results, $model)
+    public function map(Builder $builder, $results, $model)
```
Comment (kronthto) [inline src/ElasticsearchEngine.php:191]: This change makes it compatible with the new abstract method signature of `map`, but the `$builder` is not passed on to `getScoutModelsByIds` to allow Scout to actually execute the query-callback that was added in Scout 5.

**Review by kronthto — COMMENTED** [review]
I also already began working on Scout 5 support: https://github.com/kronthto/laravel-scout-elastic/commit/7b02781fa76f73c23c0288715df358423d0dec28 - I'm not finished testing it thoroughly yet.

Laravel 5.6 only suggest to use Scout 5. You can still use v4. So I think the PR would be better named "Compatibility with Scout 5" rather than "Laravel 5.6"?

File: src/ElasticsearchEngine.php (hunk 2 above)
Comment (mrabbani) [reply src/ElasticsearchEngine.php:191]: Thanks for your comment. In last commit, `$builder` is passed on to `getScoutModelsByIds` to allow Scout to actually execute the query-callback that was added in Scout 5.

**Pull request closed without merging**